code/script.js

```diff
@@ -219,107 +230,142 @@ const generateBoard = () => {
+    keep = true;
+  } else {
+    keep = false;
   }
-}
 
-// It'll filter the characters array and redraw the game board.
+  // Passing on the keep value to the filterCharacters function
+  filterCharacters(keep);
+
+}// Here the characters array is filtered and game board is redrawn
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
-    }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
+  const { category, value } = currentQuestion;
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  if (category === 'house' || category === 'hair' || category === 'type' || category === 'species') {
+    filterFunction = person => person[category] === value;
+  } else if (category === 'teacher') {
+    filterFunction = person => person.isTeacher;
+  }
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+  const positiveCharacters = charactersInPlay.filter(filterFunction);
+  const negativeCharacters = charactersInPlay.filter(person => !filterFunction(person));
 
-  // Invoke a function to redraw the board with the remaining people.
-}
+  if (keep) {
+    // Using ternary operators to shorten the way in which alerts are shown.
+    // In those cases the character is a teacher, one text is shown, in all other cases and other text is shown. This is becuase teacher has a different syntax when writing it out. 
+    alert(
+      `Yes, the character ${category === 'teacher' ? 'is a teacher' : `has ${value} ${category}`}! Keep all matching characters`
+    );
+    charactersInPlay = positiveCharacters;
+  } else {
+    alert(
+      `No, the character ${category === 'teacher' ? `isn't a teacher` : `doesn't have ${value} ${category}`}! Remove all matching characters`
+    );
+    charactersInPlay = negativeCharacters;
+  }
```

Comment:
Just a tiny note, the alert doesn't always work with the options in your dropdown. For example if the player guesses on Hufflepuff house they get the alert "No, the character doesn't have hufflepuff house!". But this is just a minor detail!

Reply:
Thank you for this! I will look into it! I've had som issuses with this, but each time I do a push I have tried checking so that it works, and those times it does work. I'm a bit perplexed as too where the issue may lay :D

Reply:
Just a follow-up question, are you certain that it wrongly alerted No on Hufflepuff? I only have one person from Hufflepuff so maybe it just isn't visible that it disappears :D ?

Reply:
Oh no, I'm so sorry for being so unclear! The alert function works just fine!! I just meant that the sentence it puts out gets a little weird for some of the options. Like when you ask about a house it says ".. doesn't have Hufflepuff house" but it should maybe be something like ".. is not in hufflepuff". Hope you understand what I mean, like I said it's just a minor detail! :D

Reply:
Haha, oh ok, now I understand!! :D